scheduler/core/components/selector/__init__.py

```diff
@@ -176,7 +176,8 @@ def score_program(self,
         # Calculate the group info and put it in the structure if there is actually group
         # info data inside it, i.e. feasible time slots for it in the plan.
         # This will filter out all GroupInfo objects that do not have schedulable slots.
-        unfiltered_group_data_map = self._calculate_group(program.root_group,
+        unfiltered_group_data_map = self._calculate_group(program,
+                                                          program.root_group,
```

Comment:
Passing program.root_group seems redundant now if we are passing program as well.

Reply:
It is a bit, but the method runs recursively, so we have to pass it some `Group`.
The `Program` is just passed in because we need it to score `Observation`s with info like whether or not the `Program` is a thesis program.